src/mscorlib/src/System/IO/Stream.cs

```diff
@@ -155,7 +185,37 @@ public virtual Task CopyToAsync(Stream destination, Int32 bufferSize, Cancellati
         // the current position.
         public void CopyTo(Stream destination)
         {
-            CopyTo(destination, _DefaultCopyBufferSize);
+            int bufferSize = _DefaultCopyBufferSize;
+            if (CanSeek)
+            {
+                long length = Length;
+                long position = Position;
+                if (length <= position) // Handles negative overflows
+                {
+                    // No bytes left in stream
+
+                    // Unlike CopyToAsync, since neither overloads of Stream.CopyTo
+                    // were virtual at the time this short-circuit optimization was
+                    // introduced, we don't need to worry about the other overload
+                    // performing nonexistent/different argument validation. We can
+                    // simply call ValidateCopyToArguments again here, and return.
+
+                    // NOTE NOTE NOTE: If the other CopyTo overload is made virtual in
```

Comment:
I would keep the two implementations consistent and just set the buffersize to 1. Saving 1 byte array allocation does not seem to be worth the amount of comments.